### PR TITLE
screencast

### LINKED ISSUE #47: mir_demo_client_screencast always fails
$ mir_demo_client_screencast
screencast_failed

$ mir_demo_client_screencast -f screencast-raw -d 0 -s 1200x900
screencast_failed

NB the -d "output id" isn't the "id" as displayed by mirout as mir_display_config_get_output() takes a 0 based index. Using "1" causes an assert.

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Enable screen capture to SHM buffers. (Fixes #47)

### CODE REVIEW

File: include/renderers/gl/mir/renderer/gl/texture_target.h [hunk 1]
```diff
@@ -34,6 +34,9 @@ class TextureTarget
     /* Bind the target as a texture suitable for rendering to via glFramebufferTexture2D */
     virtual void bind_for_write() = 0;
 
+    /// Memory backed targets need to secure the pixels after rendering
+    virtual void secure_pixels() = 0;
```
Comment (gerboland) [inline include/renderers/gl/mir/renderer/gl/texture_target.h]: My only other thought is secure_for_read(), as I expect the typical use is to read() right after.

File: src/renderers/gl/renderer.cpp [hunk 2]
```diff
@@ -217,6 +217,9 @@ void mrg::Renderer::render(mg::RenderableList const& renderables) const
     // Deleting unused textures only requires the GL context. This clean-up
     // does not affect screen contents so can happen after swap_buffers...
     texture_cache->drop_unused();
+
+    if (auto gl_error = glGetError())
+        mir::log_debug("GL error: %d", gl_error);
```
Comment (gerboland) [inline src/renderers/gl/renderer.cpp]: Useful? We can enable printing of all GL errors with MESA_DEBUG=1.

Also glGetError fetches the first in a queue of errors, see https://www.khronos.org/opengl/wiki/OpenGL_Error
which means an error printed here may not be related to the last gl function call.

It doesn't hurt though.

Comment (AlanGriffiths) [reply src/renderers/gl/renderer.cpp]: I know it is of limited use, but at least *something* gets logged to indicate an error.

Would you suggest this?

   while (auto gl_error = glGetError())
       mir::log_debug("GL error: %d", gl_error);

I'm happy to remove it though, if you feel that would be better.

Comment (gerboland) [reply src/renderers/gl/renderer.cpp]: I'm ok with keeping it in, to at least indicate an error happened.

**Pull request merged**